### PR TITLE
#140679451 : Implement HA for the event generator

### PR DESCRIPTION
**Pull request by rohitsharma04**
- Added consul based lock in eventgenerator.
- Eventgenerator does not expose a server. So there's no service registration. If this is true, then we will create another PR to do some cleanup of the eventgenerator.

### CODE REVIEW

**Comment by fraenkel** [comment]
While it is not a service, I would still add the service registration so one can easily locate the current event generator. Looking up in DNS would be easier than querying consul to find the active event generator.

File: src/autoscaler/eventgenerator/cmd/eventgenerator/eventgenerator_suite_test.go [hunk 1]
```diff
@@ -88,16 +91,31 @@ var _ = SynchronizedBeforeSuite(func() []byte {
...
+		consulRunner.Stop()
+	}
 }, func() {
 	gexec.CleanupBuildArtifacts()
 })
 
+func initConsul() {
+	consulRunner = consulrunner.NewClusterRunner(
+		consulrunner.ClusterRunnerConfig{
+			StartingPort: 9001,
```
Comment (fraenkel) [inline src/autoscaler/eventgenerator/cmd/eventgenerator/eventgenerator_suite_test.go]: How does this work when you run in parallel?

File: src/autoscaler/eventgenerator/cmd/eventgenerator/eventgenerator_test.go [hunk 2]
```diff
@@ -1,42 +1,112 @@
...
+			runner.Start()
+
+			Eventually(runner.Session.Buffer, 2*time.Second).Should(gbytes.Say(runner.acquiredLockCheck))
+			Eventually(runner.Session.Buffer, 2*time.Second).Should(gbytes.Say("eventgenerator.started"))
+
+			consulRunner.Reset()
+		})
+
+		It("exits with failure", func() {
+			Eventually(runner.Session.Buffer, 4*time.Second).Should(gbytes.Say("exited-with-failure"))
```
Comment (fraenkel) [inline src/autoscaler/eventgenerator/cmd/eventgenerator/eventgenerator_test.go]: If you flip these, the check for exited becomes an Expect.

File: src/autoscaler/eventgenerator/cmd/eventgenerator/eventgenerator_suite_test.go (hunk 1 above)
Comment (rohitsharma04) [inline src/autoscaler/eventgenerator/cmd/eventgenerator/eventgenerator_suite_test.go]: @fraenkel  should I also update the metricscollector to run in parallel?

Comment (fraenkel) [inline src/autoscaler/eventgenerator/cmd/eventgenerator/eventgenerator_suite_test.go]: yes, we should make all of our units run in parallel when possible

**Pull request merged**